test/integration/stats/applications_test.rb

```diff
@@ -14,19 +14,39 @@ def setup
     login_provider @provider
   end
 
-  def test_show
-    get admin_buyers_stats_application_path(id: @application.id)
-    assert_response :success
-  end
+  context 'with access to all services' do
+    setup do
+      User.any_instance.expects(:has_access_to_all_services?).returns(true).at_least_once
+    end
+
+    should '#show with member permission' do
+      User.any_instance.expects(:member_permission_service_ids).never
+      get admin_buyers_stats_application_path(id: @application.id)
+      assert_response :success
+    end
 
-  def test_show_without_access_to_all_services
-    User.any_instance.expects(:has_access_to_all_services?).returns(false).at_least_once
-    get admin_buyers_stats_application_path(id: @application.id)
-    assert_response :forbidden
+    should '#show without member permission' do
```

Comment:
this and `show with member permission` appear to be absolutely the same code

Reply:
Yes indeed